src/main/java/dev/slimevr/vr/trackers/IMUTracker.java

```diff
@@ -101,6 +133,12 @@ public void tick() {
 				calculateLiveMagnetometerCorrection();
 			}
 		}
+		if(movementFilterTickCount != 0){
+			previousRots.addLast(new Quaternion(rotQuaternion));
```

Comment:
Considering the tick count should be a fixed number, maybe this could be an array of quaternions that are looped over instead?

Reply:
Like a circle buffer?

Reply:
Also changed to Quaternion.clone here instead of creating a new Quaternion

Reply:
Yeah, something like that, since it's happening hundreds of times per second, I think it'd be smart to try to keep allocation down a bit

Reply:
What do you think about that? http://www.museful.net/2012/software-development/circulararraylist-for-java

Reply:
Alright. I'm using a circle buffer now. Performance should be better. _This is also what my dad said would be the most optimized for this kind of usage but I was too lazy to implement it._